### ISSUE TITLE
Contents of spreadsheets disappear until the server is rebooted

### ISSUE THREAD

**Issue by Armael** [cryptpad/cryptpad#1699]
### Description

At Deuxfleurs (deuxfleurs.fr) we are observing a fairly severe regression on our self-hosted Cryptpad instance, running version 2024.9.0.

We are observing that, after some time, spreadsheets **in the "Fast" co-editing mode** silently become "corrupted" and appear as blank. Spreadsheets in "Strict" co-editing mode do not seem to be impacted. Several of our users also have been reporting this issue affecting their personal documents.

Restarting the cryptpad server **temporarily solves the issue**: the contents of the affected spreadsheets become visible again. Of course, this is not a tractable solution in the long term, and we are also worried that operating in this "degraded" mode may eventually lead to data corruption.

Each time we had to reboot, Cryptpad was using more RAM than average right before. Something like 100 MiB more than the usual, which is around 150 MiB in our servers.
![image](https://github.com/user-attachments/assets/8071b39c-6397-42c8-bfef-973e58812aff)

This issue only happens on spreadsheets, and not on any other type of documents.

We are happy to provide more info about our deployment if this can help debugging the issue.

A side question: as a workaround, would it be safe in the mean time to downgrade our Cryptpad instance from 2024.9.0 to 2024.6.0 (which we were previously running without issues), without data loss?

### Steps to reproduce

At this time we do not unfortunately know of a reliable way of triggering the issue. (We have recently started a test cryptpad instance in parallel of our production server, but so far the issues have only appeared in the production instance).

### Expected behavior

For the contents of spreadsheets to not disappear.

### Which operating system are you using?

Linux/BSD/UNIX

### Extensions

N/A

### Version

2024.9.0

**Comment by Armael** [comment]
>A side question: as a workaround, would it be safe in the mean time to downgrade our Cryptpad instance from 2024.9.0 to 2024.6.0 (which we were previously running without issues), without data loss?

Any ideas regarding this last question? It would be useful for us to know what to expect if we try to rollback from 2024.9.0 to 2024.6.0 (which for now seems to be our best bet to mitigate the issue).

**Comment by davidbenque** [comment]
Hi @Armael thank you for this report and sorry for taking a while to respond. Rolling back to 2024.6.0 is probably the best course of action right now until the root cause of the issue is known.

We have not had any similar reports so perhaps you could share a little more about the system you are using? (operating system, web server, etc)

**Comment by Armael** [comment]
Hi! Thanks for the answer. In the meantime we've investigated the issue a bit, so let me report our findings so far:

- Rolling back to 2024.6.1 worked without apparent issues, so this gives us a "known good" configuration to fall back to.
- I noticed another issue: importing a `.xls` document in an onlyoffice spreadsheet was not working on our cryptpad instance.
  - Looking at the JS console showed a bunch of CORS errors, and I could solve the issue by adding a `Access-Control-Allow-Origin *` header in our reverse proxy, which made "Import" work again.
  - I'm a bit confused as to why this is needed though. In [the sample nginx config](https://github.com/cryptpad/cryptpad/blob/main/docs/example.nginx.conf) the header isn't added by nginx, but it is added (along many others) in the [advanced nginx config](https://github.com/cryptpad/cryptpad/blob/main/docs/example-advanced.nginx.conf). I thus assumed that in the "simple case" (where the reverse proxy just forwards all the requests, which is what we do) the `node` server takes care of producing all the required headers. But this doesn't seem to be the case: the `node` server seems to be producing some headers (e.g. `Cross-Origin-Resource-Polity`, etc) but not this one.

I don't know if the initial issue (disappearing spreadsheets' contents) is related to this missing header. To test this hypothesis, we are currently running our cryptpad instance back *on version 2024.9.0 but with the header added*, and we will see if the issue comes back.

Now, to give a bit more context on our deployment (which is admittedly somewhat nonstandard):
- we build/package `cryptpad` as a docker image, that gets deployed by Nomad (our orchestrator) ([corresponding nomad job description](https://git.deuxfleurs.fr/Deuxfleurs/nixcfg/src/commit/1a739636ca239308c890244c41c16fcd2a351b7b/cluster/prod/app/cryptpad/deploy/cryptpad.hcl))
  - the docker image is currently built using some nix invocations: https://git.deuxfleurs.fr/Deuxfleurs/nixcfg/src/branch/main/cluster/prod/app/cryptpad/build 
  - if the issue persists, we could instead try to build the docker image using a Dockerfile that follows a more standard installation procedure (e.g. using the `install-onlyoffice.sh` script) ([Dockerfile](https://git.deuxfleurs.fr/Deuxfleurs/nixcfg/src/commit/1a739636ca239308c890244c41c16fcd2a351b7b/cluster/prod/app/cryptpad/build_docker/Dockerfile) already tested on a debug instance)
- our reverse proxy is [tricot](https://git.deuxfleurs.fr/Deuxfleurs/tricot), which reads its configuration from [Nomad tags](https://git.deuxfleurs.fr/Deuxfleurs/nixcfg/src/commit/1a739636ca239308c890244c41c16fcd2a351b7b/cluster/prod/app/cryptpad/deploy/cryptpad.hcl#L62).
- the machine runs NixOS (but this is most likely not relevant here since deployment uses docker+nomad)
- the cryptpad nomad service doesn't look particularly resource-constrained, in particular it didn't use more than 400Mo of RAM over the last two months, out of its allocated 1G.

**Comment by Armael** [comment]
> I thus assumed that in the "simple case" (where the reverse proxy just forwards all the requests, which is what we do) the `node` server takes care of producing all the required headers. But this doesn't seem to be the case: the `node` server seems to be producing some headers (e.g. `Cross-Origin-Resource-Polity`, etc) but not this one.

Sorry, I just realize that this is somewhat inaccurate. In fact, `node` does produce a `Access-Control-Allow-Origin` header, but for the main domain, whereas it looks like we need it (at least) for the *sandbox* domain (which is what the "advanced" config does)?

**Comment by fbohmann** [comment]
i have exactly the same issue on my 2024.9.0 instance pad.abraum.de, running on a kvm virtual machine, installed from git, no docker. i do not really know how to debug this, but i will try the advanced config file. import of .xls also not working here.

**Comment by Armael** [comment]
@fbohmann fixing the CORS-related headers in the reverse proxy (as I described above) fixed the import issue for us. Additionally, since the headers change we've (for now) not seen the issue reappear. So it would be interesting to see if the same fix could work for you.

**Comment by fruit79** [comment]
Hello same issue here with the 5.3.0 version of cryptpad for yunohost

**Comment by sudoaza** [comment]
Hi, getting the same in version 2024.6.0. This is quite annoying since during reboot some users appear to have lost other documents they were working on. No obvious error message.

**Comment by Armael** [comment]
We've again noticed this issue, or at least one with similar effects.
I currently have two onlyoffice spreadsheets (originally imported from .xslx files), with the following behavior:
- they loaded correctly the first few times they were opened on a given computer/browser
- the next times, they just load into a blank spreadsheet.
- using private browsing, I can get the spreadsheet to load properly the first time, but after refreshing (F5) one or two times it becomes blank
- interestingly, *using the Firefox Page Inspector makes the page load properly again!* (ie starting from the spreadsheet its it "bugged" state (blank), opening the inspector and reloading (F5) makes the spreadsheet load again. closing the inspector and reloading (F5) makes it blank again). 
- Using the Chromium Page Inspector does not change anything.

I'm not sure this is exactly the same issue as the one we were observing before, but the symptoms are close enough that I thought I'd report it here.

**Comment by Armael** [comment]
(I'm still seeing some strange CORS-related issues in the console, that I can't explain yet; I will investigate and report back...)

**Comment by Armael** [comment]
Sorry for the noise: the issue was on our end. Our reverse proxy was not correctly applying CORS headers for 3xx (in particular 304) responses. This is now fixed and seems to make the issue (https://github.com/cryptpad/cryptpad/issues/1699#issuecomment-2509641070) disapper.
